source/slang/slang.cpp

```diff
@@ -372,16 +372,47 @@ SlangResult CompileRequest::loadFile(String const& path, ISlangBlob** outBlob)
 
 RefPtr<Expr> CompileRequest::parseTypeString(TranslationUnitRequest * translationUnit, String typeStr, RefPtr<Scope> scope)
 {
-    Slang::RefPtr<Slang::SourceFile> srcFile = sourceManager->createSourceFile(String("type string"), typeStr);
+    // Create a SourceManager on the stack, so any allocations for 'SourceFile'/'SourceView' etc will be cleaned up
+    SourceManager localSourceManager;
+    localSourceManager.initialize(sourceManager);
+        
+    Slang::RefPtr<Slang::SourceFile> srcFile(localSourceManager.createSourceFile(String("type string"), typeStr));
     
+    // We'll use a temporary diagnostic sink  
     DiagnosticSink sink;
-    sink.sourceManager = sourceManager;
+    sink.sourceManager = &localSourceManager;
+
+    // RAII type to make make sure current SourceManager is restored after parse.
+    // Use RAII - to make sure everything is reset even if an exception is thrown.
+    struct ScopeReplaceSourceManager
+    {
```

Comment:
It would be great if we didn't need to modify the internal state of the `CompileRequest` here (perhaps creating a separate "sub" compile request for parsing the type string), but it is fine as workaround for now.